File: lc-dc-sep/lc-dc-sep5/solution.py
# Problem: 2028. Find Missing Observations

class Solution:
    def missingRolls(self, rolls: List[int], mean: int, n: int) -> List[int]:
        size = len(rolls) + n
        rollsSum = sum(rolls)
        missing = mean * size - rollsSum
        res = []

        if missing < n or missing > 6*n:
            return res

        q = missing // n
        r = missing % n

        for i in range(n):
            res.append(q)

        for i in range(r):
            res[i] += 1

        return res

File: lc-dc-sep/lc-dc-sep5/solution.cpp
// Problem: 2028. Find Missing Observations

class Solution {
public:
    vector<int> missingRolls(vector<int>& rolls, int mean, int n) {
        int len = rolls.size() + n;
        int sum = accumulate(rolls.begin(), rolls.end(), 0);
        int missing = mean * len - sum;
        vector<int> res;
        
        if(missing < n || missing > 6*n) return {};

        int quo = missing / n, rem = missing % n;
        
        for(int i=0; i<n; i++){
            res.push_back(quo);
        }

        for(int i=0; i<rem; i++){
            res.at(i)++;
        }

        return res;
    }
};